mewarpx/mewarpx/utils_store/json_util.py

```diff
@@ -0,0 +1,57 @@
+from collections import OrderedDict
+import datetime
+import json
+
+import numpy as np
+import periodictable
+
+from mewarpx import diags, emission
+
+
+class MEWarpXEncoder(json.JSONEncoder):
+    """A JSON encoder for MEWarpX objects."""
+
+    def default(self, object):
+        """Default operations if JSON library fails to serialize an object."""
+        if isinstance(object, datetime.datetime):
+            return object.strftime((r"%m/%d/%Y_%H:%M:%S:%f"))
+
+        elif isinstance(object, OrderedDict):
+            return object
+
+        elif isinstance(object, (bytes, bytearray)):
+            return object.decode("utf-8")
+
+        # only include name of element
+        elif isinstance(object, (periodictable.core.Isotope,
+                                 periodictable.core.Element)):
+            return object.name
+
+        # leave out diags to avoid circular serialization error
+        elif isinstance(object, diags.WarpXDiagnostic):
+            return str(object)
+
+        # remove very long and unneeded attributes
+        elif isinstance(object, emission.ArbitraryEmitter2D):
```

Comment:
Perhaps better to use
```suggestion
        elif isinstance(object, emission.Emitter):
```
since then this functionality will also work if we make a `ArbitraryEmitter3D` at some point.